eclipse-collections/src/main/java/org/eclipse/collections/impl/list/primitive/IntInterval.java

```diff
@@ -1007,4 +1016,82 @@ public int next()
             throw new NoSuchElementException();
         }
     }
+
+    private static final class IntIntervalSpliterator implements Spliterator.OfInt
+    {
+        private int current;
+        private final int to;
+        private final int step;
+        private final boolean isAscending;
+
+        private IntIntervalSpliterator(int from, int to, int step)
+        {
+            this.to = to;
+            this.step = step;
+            this.current = from;
+            this.isAscending = from <= to;
+        }
+
+        @Override
+        public Comparator<? super Integer> getComparator()
+        {
+            return  new Comparator<Integer>()
+            {
+                @Override
+                public int compare(Integer o1, Integer o2)
+                {
+                    if (IntIntervalSpliterator.this.isAscending)
+                    {
+                        return o1 - o2;
+                    }
+                    return o2 - o1;
+                }
+            };
```

Comment:
It would be better not to construct a comparator every time the method is called. Why not just return the constants java.util.Comparator#naturalOrder and java.util.Comparator#reverseOrder?